README.md

```diff
@@ -23,6 +23,30 @@ TODO, read the [mli](./lib/piaf.mli) file for now.
 There's an example of using Piaf's API in [bin/carl.ml](./bin/carl.ml), an
 implementation of a subset of curl, in caml.
 
+## Development
+
+There's two ways to get a development environemnt up and running. If you have (or don't mind getting) [`nix`](https://nixos.org/nix/manual/) installed, the repository includes scripts to set up a sandbox. Otherwise you can use [`opam`](https://opam.ocaml.org/) to install the necessary dependencies globally.
+
+### Option 1) Setting up the sandbox
+
+Assuming [`nix` has been installed](https://nix.dev/tutorials/install-nix.html) and set up, run `nix-shell --pure` in the repository root. Once it's done building, you should have the development environment set up!
+
+### Option 2) Setting up opam
+
+For this approach you'll need to [install `opam`](https://opam.ocaml.org/doc/Install.html)  and set it up with a switch using `ocaml` >= 4.06. Once that's done, run `opam pin . --deps-only` to install the dependencies.
```

Comment:
I think this actually only works since OCaml 4.08

Reply:
Ah, I just took that from the opam file. I can test to make sure though.

Reply:
maybe the opam file should also be updated. I haven't looked at it in a while

Reply:
Oh, actually I can't test, because I have gcc 10 and nothing below 4.09.1 works for me...

Reply:
Let's say 4.08 just to be safe though

Reply:
```suggestion
For this approach you'll need to [install `opam`](https://opam.ocaml.org/doc/Install.html)  and set it up with a switch using `ocaml` >= 4.08. Once that's done, run `opam pin . --deps-only` to install the dependencies.
```